csv-extractor/csv_extractor.py

```diff
@@ -10,7 +10,7 @@
 from operator import itemgetter
 from typing import Dict
 from collections import defaultdict
-
+from queue import Queue
```

Comment:
It is better to use collections.deque directly.
the queue.Queue used brings a lot of extra functionality that you don't use, and don't really want.

For example queue.Queue.put() when it will exceed maxsize will block! The current code insert 1 item at a time, and then check, but if you later put two items on the queue inside the same loop you are suddenly introducing a fatal flaw.

Your code would be simpler and safer, collections.deque is already threadsafe.

Using a list would probably not be any slower either. You are appending at the end, and the list with grow.